sql/size_utils.sql

```diff
@@ -126,151 +184,280 @@ BEGIN
+   FROM
+   _timescaledb_internal.hypertable_chunk_local_size ch 
+   WHERE
+      ch.schema_name = schema_name_in 
+      AND ch.table_name = table_name_in 
+; 
+END;
 $BODY$;
 
+---should return same information as chunk_local_size--
+CREATE OR REPLACE FUNCTION _timescaledb_internal.chunk_remote_size(
+    schema_name_in name,
+    table_name_in name)
+RETURNS TABLE (
+    chunk_id    integer,
+    chunk_name  NAME,
+    heap_bytes bigint,
+    index_bytes bigint,
+    toast_bytes bigint,
+    total_bytes bigint,
+    node_name NAME)
+LANGUAGE PLPGSQL STABLE STRICT AS
+$BODY$
+BEGIN
+    RETURN QUERY
+    SELECT
+        entry.chunk_id,
+        entry.chunk_name,
+        entry.table_bytes AS table_bytes,
+        entry.index_bytes AS index_bytes,
+        entry.toast_bytes AS toast_bytes,
+        entry.total_bytes AS total_bytes,
+        srv.node_name
+    FROM (
+        SELECT
+            s.node_name,
+            _timescaledb_internal.ping_data_node (s.node_name) AS node_up
+        FROM
+            _timescaledb_catalog.hypertable AS ht,
+            _timescaledb_catalog.hypertable_data_node AS s
+        WHERE
+            ht.schema_name = schema_name_in
+            AND ht.table_name = table_name_in
+            AND s.hypertable_id = ht.id
+         ) AS srv
+    LEFT OUTER JOIN LATERAL _timescaledb_internal.data_node_chunk_info(
+    CASE WHEN srv.node_up THEN
+        srv.node_name
+    ELSE
+        NULL
+    END , schema_name_in, table_name_in) entry ON TRUE;
+END;
+$BODY$;
 
--- Get relation size of hypertable
--- like pg_relation_size(hypertable)
+-- Get relation size of the chunks of an hypertable
+-- like pg_relation_size
 -- (https://www.postgresql.org/docs/9.6/static/functions-admin.html#FUNCTIONS-ADMIN-DBSIZE)
 --
 -- main_table - hypertable to get size of
 --
 -- Returns:
--- table_size         - Pretty output of table_bytes
--- index_bytes        - Pretty output of index_bytes
--- toast_bytes        - Pretty output of toast_bytes
--- total_size         - Pretty output of total_bytes
+-- chunk_id                      - Timescaledb id of a chunk
+-- chunk_table                   - Table used for the chunk
+-- partitioning_columns          - Partitioning column names
+-- partitioning_column_types     - Type of partitioning columns
+-- partitioning_hash_functions   - Hash functions of partitioning columns
+-- ranges                        - Partition ranges for each dimension of the chunk
+-- heap_bytes                   - Disk space used by main_table
+-- index_bytes                   - Disk space used by indexes
+-- toast_bytes                   - Disk space of toast tables
+-- total_bytes                   - Disk space used in total
 
-CREATE OR REPLACE FUNCTION hypertable_relation_size_pretty(
+CREATE OR REPLACE FUNCTION chunk_detailed_size(
     main_table              REGCLASS
 )
-RETURNS TABLE (table_size  TEXT,
+RETURNS TABLE (chunk_id INT,
```

Comment:
I think the schema name is redundant here. We have that information in the "chunks" view. This is just supposed to give size information for chunks. We identify it by chunk_id and/or name.

Reply:
I don't think the function should return the internal ID since it doesn't mean anything to the user. The canonical identifier is schema+name, which I think is what we should use when we don't use OIDs. (Also because this is what you need to resolve the OID.)

Another reason for not returning the internal ID is because it is inconsistent with other functions. For instance, neither `hypertable_detailed_size` nor `indexes_detailed_size` return such IDs.

Reply:
Yeah, I don't mind that internal functions (under `_timescaledb_internal`) returns IDs if it is needed for some reason, but public functions should only deal with official identifiers such as schema+name or OIDs.

Reply:
@WireBaron Good catch on the chunk_id. Removing it from the external function.